docs/deployment/contract-deployment-options.md

```diff
@@ -0,0 +1,114 @@
+
+- `version?` - `string`: The Contract version you want to deploy. Defaults to latest version of [lsp-smart-contracts](https://github.com/lukso-network/lsp-smart-contracts) library.
+- `byteCode?` - `string`: Custom bytecode to be deployed.
+- `deployProxy?` - `boolean`: Whether the contract should be deployed using a proxy contract implementation (eg: [EIP1167](https://eips.ethereum.org/EIPS/eip-1167)). Defaults to true.
+- `libAddress?` - `string`: The Address of a Base Contract to be used in deployment as implementation behind a proxy contract (eg: [EIP1167](https://eips.ethereum.org/EIPS/eip-1167)).
+
+
+This is an optional parameter and so may be ommitted. If no contract deployment options are specified, `LSPFactory` will deploy a proxy contract which defers to a base contract implementation as per the [EIP1167](https://eips.ethereum.org/EIPS/eip-1167) standard.
+
+If you do not want your contract to use proxy deployment you can set `deployProxy` to `false`. This will deploy a 'full' contract rather than using a proxy deployment.
+
+:::info Info
+LSPFactory stores base contract addresses for different versions [internally](https://github.com/lukso-network/tools-lsp-factory/blob/main/src/versions.json) and uses the latest available version if no version is specified.
+:::
+
+#### Custom Bytecode
+
+By passing the `bytecode` parameter you can specify the bytecode you want your contract to use. Unless `deployProxy` is set to `false`, LSPFactory will deploy a proxy contract which will defer to a base contract where your bytecode is deployed.
+
+For example, you could deploy a Universal Profile with a KeyManager which uses your custom bytecode as a base contract implementation:
+
+```javascript title="Deploying a Universal Profile with a custom KeyManager base contract"
+lspFactory.LSP3UniversalProfile.deploy({...}, {
+    version: '0.4.1'
+    KeyManager: {
+        bytecode: '0x...',
+    }
+})
+```
+
+Or if you want your custom bytecode to be deployed at the KeyManager contract itself rather than at the base contract implementation you can set `deployProxy` to false:
+
+```javascript title="Deploying a Universal Profile with a custom KeyManager base contract"
+lspFactory.LSP3UniversalProfile.deploy({...}, {
+    version: '0.4.1'
+    KeyManager: {
+        bytecode: '0x...',
+        deployProxy: false
+    }
+})
+```
+
+### Custom Universal Profile Deployment
+
+A [Universal Profile](../classes/lsp3-universal-profile) is composed of three different contracts. By passing the global `version` parameter you can set the version for all contracts at once.
+
+```javascript title="Deploying a Universal Profile with all contracts at version 0.4.1"
+lspFactory.LSP3UniversalProfile.deploy({...}, {
+    version: '0.4.1'
+})
+```
+
+<!-- :::info Infos -->
+
+**LSPFactory also allows contracts to be individually customisable. You can set the version per contract which will take presidence over the global version:**
```

Comment:
_typo:_

"LSPFactory also allows contracts to be individually customisable. You can set the version per contract, which will take ~~presidence~~ **precedence** over the global version:"